File: SustitucionMOAModel/Entities/Consulta.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SustitucionMOAModel.Entities
{
    public class Consulta
    {
        [Key]
        public int Id { get; set; }
        public string CodigoCorredor { get; set; }
        public string RazonSocialCorredor { get; set; }
        public string CodigoProveedor { get; set; }
        public string RazonSocialProveedor { get; set; }
        public int Categoria_Id { get; set; }
        public int? SubCategoria_Id { get; set; }
        public string Asunto { get; set; }
        public int EstadoConsulta_Id { get; set; }
        public DateTime FechaCreacion { get; set; }
        public DateTime FechaUltimaModificacion { get; set; }
        public int Usuario_Id { get; set; }
        public int UsuarioInterno_Id { get; set; }

        [ForeignKey("Usuario_Id")]
        public virtual Usuario Usuario { get; set; }

        [ForeignKey("Categoria_Id")]
        public virtual Categoria Categoria { get; set; }

        [ForeignKey("SubCategoria_Id")]
        public virtual SubCategoria SubCategoria { get; set; }

        [ForeignKey("EstadoConsulta_Id")]
        public virtual EstadoConsulta EstadoConsulta { get; set; }

        [Required]
        public virtual ConsultaDetalle Detalle { get; set; }

        public virtual ICollection<Comentario> Comentarios { get; set; }

        public Consulta() { }
    }
}

File: SustitucionMOA/src/app/consulta/consulta.ts
import { DirOrden } from "../common/enums/DirOrden";
import { Archivo } from "../common/models/archivo";
import { Usuario } from "../usuario/usuario";

export interface Consulta {
    Id;
    CodigoCorredor;
    RazonSocialCorredor;
    CodigoProveedor;
    RazonSocialProveedor;
    CategoriaId;
    SubCategoriaId?;
    Asunto;
    Material;
    EstadoConsultaId;
    Material_Id;
    FechaCreacion;
    FechaUltimaModificacion;
    UsuarioId;
    UsuarioActualId;
    UsuarioInternoId;
    FechaVtoReapertura;


    Fecha;
    ComprobanteNo;
    OtroComprobanteNo;
    ContratoNo;
    Importe;
    Impuesto;
    BolsaEmisoraOblea;
    OrdenId;
    PatenteChasis;

    Categoria;
    SubCategoria?;
    EstadoConsulta;
    CausaConsulta;
    Comentarios: Comentario[];

    DiasReclamo;
    Usuario?;
    PuedeReabrir;

    RelacionadaPorCodigo?: boolean;
    GeneradaInternamente?: boolean;
    GeneradaExternamente?: boolean;
    GeneradaPorUsuarioSesion?: boolean;

    MailUsuarioIniciaConsulta?: string;
    Rubro: string;
}

export interface EstadoConsulta {
    Id: number;
    Code: string;
    Descripcion: string;
    Color?: string;
    Cantidad?: number;
}

export interface Categoria {
    Id;
    Code;
    Nombre;
    Cantidad;
}

export interface Subcategoria {
    Id;
    Code?;
    Nombre?;
    CategoriaId?;
    Cantidad?;
}

export interface Causa {
    Id;
    Nombre;
}

export interface Materiales {
    MaterialId;
    Descripcion;
}

export class Comentario {
    Consulta_Id?: any;
    consulta_Id: any;
    Detalle: string;
    Fecha: any;
    FechaRecordado: any;
    UsuarioId?: number;
    Id?: number;
    Usuario?: Usuario;
    Recordado?: boolean;
    CreadorInterno?: boolean;
    Archivos?: Archivo[];
    ComentarioRecordados?: any;
}
export class ReclamoImpositivo {
    Dni;
    RazonSocialEmpresa;
    RazonSocialProveedor;
    Cuit;
    Vinculo;
    Reclamos: Array<Reclamo>;
    Lugar;
}
export interface Reclamo {
    Fecha;
    Certificado;
    Importe;
}

export class Destinatario {
    Campo: string;
    Mail: string;
    UsuarioId: number;
    NombreTipoUsuario;
}

export enum OpcionFiltroAsociadaCreacion {
    Externa = "Externa",
    PorMOA = "Por MOA",
    PorUsuario = "Propias",
}

export function obtenerOpcionesFiltroPorCreacion(): Array<{ key: OpcionFiltroAsociadaCreacion, value: OpcionFiltroAsociadaCreacion }> {
    return [
        { key: OpcionFiltroAsociadaCreacion.Externa, value: OpcionFiltroAsociadaCreacion.Externa },
        { key: OpcionFiltroAsociadaCreacion.PorMOA, value: OpcionFiltroAsociadaCreacion.PorMOA },
        { key: OpcionFiltroAsociadaCreacion.PorUsuario, value: OpcionFiltroAsociadaCreacion.PorUsuario },
    ]
}
export interface ReqListadoConsultaDto {
    page: number;
    pageSize: number;
    orderBy: keyof Consulta;
    dirOrden: DirOrden;
    filtros?: Record<keyof Consulta, any>;
}